modules/index/src/main/java/org/apache/ignite/internal/index/ChangeIndexStatusTask.java

```diff
@@ -333,8 +335,26 @@ private <T> CompletableFuture<T> inBusyLocks(Supplier<CompletableFuture<T>> supp
 
         try {
             return supplier.get();
+        } catch (Throwable t) {
```

Comment:
We should not swallow `Error`s (which are not `AssertionError`). For them, we should probably log them and rethrow them after this.

Reply:
I didn't fully understand you.
This method will return a fallen future with an error that it looks more correct, and processing this error, including logging, should not happen here. There is already processing in **org.apache.ignite.internal.index.ChangeIndexStatusTask#start**.